include/boost/numeric/ublas/tensor/detail/extents_functions.hpp

```diff
@@ -240,7 +239,7 @@ template <class ExtentsType>
 inline 
 constexpr bool is_vector(ExtentsType const &e) {
   
-  static_assert(detail::is_extents<ExtentsType>::value, "boost::numeric::ublas::is_vector() : invalid type, type should be an extents");
+  static_assert(is_extents_v<ExtentsType>, "boost::numeric::ublas::is_vector() : invalid type, type should be an extents");
 
   using size_type = typename ExtentsType::size_type;
   using value_type = typename ExtentsType::value_type; 
```

Comment:
I like the algorithm but I recommend stylisticly to have sth like. 
```cpp
constexpr auto greater_one = [](auto a){ return a>1u; }; 
constexpr auto equal_one   = [](auto a){ return a==1u; }; 

if      (e.size() == 0u) return false;
else if (e.size() == 1u) return e[0]>1u;
return /* as you have */
```

Reply:
ok